modules/sdk/gradle-plugins/src/com/liferay/gradle/plugins/LiferayWebAppPlugin.java

```diff
@@ -155,12 +179,90 @@ protected void configureTaskBuildCssRootDirs(BuildCssTask buildCssTask) {
 		buildCssTask.setRootDirs(getWebAppDir(project));
 	}
 
+	protected void configureTaskDirectDeploy(
+		Project project, LiferayExtension liferayExtension) {
+
+		DirectDeployTask directDeployTask =
+			(DirectDeployTask)GradleUtil.getTask(
+				project, DIRECT_DEPLOY_TASK_NAME);
+
+		configureTaskDirectDeployAppServerDeployDir(
+			directDeployTask, liferayExtension);
+		configureTaskDirectDeployAppServerLibGlobalDir(
+			directDeployTask, liferayExtension);
+		configureTaskDirectDeployAppServerPortalDir(
+			directDeployTask, liferayExtension);
+		configureTaskDirectDeployAppServerType(
+			directDeployTask, liferayExtension);
+		configureTaskDirectDeployWebAppFile(directDeployTask);
+		configureTaskDirectDeployWebAppType(directDeployTask);
+	}
+
+	protected void configureTaskDirectDeployAppServerDeployDir(
+		DirectDeployTask directDeployTask, LiferayExtension liferayExtension) {
+
+		if (directDeployTask.getAppServerDeployDir() == null) {
+			directDeployTask.setAppServerDeployDir(
+				liferayExtension.getAppServerDeployDir());
+		}
+	}
+
+	protected void configureTaskDirectDeployAppServerLibGlobalDir(
+		DirectDeployTask directDeployTask, LiferayExtension liferayExtension) {
+
+		if (directDeployTask.getAppServerLibGlobalDir() == null) {
+			directDeployTask.setAppServerLibGlobalDir(
+				liferayExtension.getAppServerLibGlobalDir());
+		}
+	}
+
+	protected void configureTaskDirectDeployAppServerPortalDir(
+		DirectDeployTask directDeployTask, LiferayExtension liferayExtension) {
+
+		if (directDeployTask.getAppServerPortalDir() == null) {
+			directDeployTask.setAppServerPortalDir(
+				liferayExtension.getAppServerPortalDir());
+		}
+	}
+
+	protected void configureTaskDirectDeployAppServerType(
+		DirectDeployTask directDeployTask, LiferayExtension liferayExtension) {
+
+		if (Validator.isNull(directDeployTask.getAppServerType())) {
+			directDeployTask.setAppServerType(
+				liferayExtension.getAppServerType());
+		}
+	}
+
+	protected void configureTaskDirectDeployWebAppFile(
+		DirectDeployTask directDeployTask) {
+
+		if (directDeployTask.getWebAppFile() != null) {
+			return;
+		}
+
+		War warTask = (War)GradleUtil.getTask(
```

Comment:
This variable should probably be called `war`, but there's still a bit of inconsistency somewhere else in the code. I'm planning on fixing this kind of problem everywhere in the next PR :blush: